account/recovery.go

```diff
@@ -523,3 +525,53 @@ func GenerateRecoveryKeys(ctx context.Context, accountTarget uint32,
 	}
 	return acctKeys, nil
 }
+
+// AdvanceAccountDerivationIndex asserts that the internal wallet has at least
+// the given minimum index for deriving account keys.
+func AdvanceAccountDerivationIndex(ctx context.Context, minimumIndex uint32,
+	wallet lndclient.WalletKitClient, chainParams *chaincfg.Params) error {
+
+	allAccounts, err := wallet.ListAccounts(
+		ctx, "", walletrpc.AddressType_UNKNOWN,
+	)
+	if err != nil {
+		return fmt.Errorf("error listing accounts: %v", err)
+	}
+
+	poolAccountsPath := fmt.Sprintf("m/%d'/%d'/%d'",
+		keychain.BIP0043Purpose, chainParams.HDCoinType,
+		poolscript.AccountKeyFamily)
+
+	var poolAccountsAccount *walletrpc.Account
+	for idx, acct := range allAccounts {
+		if acct.DerivationPath == poolAccountsPath {
+			poolAccountsAccount = allAccounts[idx]
```

Comment:
nit: can we move the `poolAccountsAccount.ExternalKeyCount > minimumIndex` and do an early return

```|
if acct.DerivationPath == poolAccountsPath && acct.ExternalKeyCount > minimumIndex {
    // log + return
}

for {
    // ...
```